AEPMessaging/Sources/Messaging+PublicAPI.swift

```diff
@@ -64,4 +94,73 @@ import UserNotifications
 
         MobileCore.dispatch(event: event)
     }
+
+    // MARK: - Private Helper Methods
+
+    /// Determines whether the user's response to a notification has caused the application to open
+    ///
+    /// This method analyzes the registered categories and notification action buttons of the application
+    /// and determines if the application was opened based on the action performed by the user. The result is provided through the `completion` closure.
+    ///
+    /// - Parameters:
+    ///   - response: The user's response to a notification, represented by a `UNNotificationResponse` object.
+    ///   - completion: A closure that takes a `Bool` parameter indicating whether the application was opened or not. This closure is invoked asynchronously once the determination is made.
+    ///
+    /// - Note: The completion handler is invoked asynchronously, so any code relying on the result should be placed within the completion handler or called from there.
+    private static func hasApplicationOpenedForResponse(_ response: UNNotificationResponse, completion: @escaping (Bool) -> Void) {
```

Comment:
any particular reason we implemented as a callback rather than just having this method return a bool?

Reply:
Internally within this method we call iOS API

```
 UNUserNotificationCenter.current().getNotificationCategories { categories in
  }
```
which makes the code asynchronous, hence the callback